### DISCUSSION TITLE
xmlsec error

### DISCUSSION THREAD

**Discussion by idlebrain001** [lsh123/xmlsec#713] [Q&A] [answered]
I am trying to setup SAML based SSO using mod_auth_mellon on apache HTTP running RHEL 7.9, but getting these errors in apache logs.  Can someone please help me here?  I have all the libraries and LD_LIBRARY_PATH is also defined.  

{ "timestamp":"Wed Oct 25 01:48:50 2023TWed Oct 25 01:48:50 2023.2023-10-25 01:48:50Z", "function" : "[core:notice]" , "process" : "[pid 18269:tid 139762094909312]" , "message" : "AH00094: Command line: '/apps/apache/2.4.57/bin/httpd -f /apps/apache/conf/httpd.conf'" , "referer" : },

func=xmlSecCryptoDLLibraryCreate:file=dl.c:line=146:obj=lt_dlopenext:subj=unknown:error=7:io function failed:filename=libxmlsec1-openssl

func=xmlSecCryptoDLGetLibraryFunctions:file=dl.c:line=498:obj=unknown:subj=xmlSecCryptoDLLibraryCreate:error=1:xmlsec library function failed:crypto=openssl

func=xmlSecCryptoDLLoadLibrary:file=dl.c:line=449:obj=unknown:subj=xmlSecCryptoDLGetLibraryFunctions:error=1:xmlsec library function failed:
 

(process:18274): Lasso-CRITICAL **: 01:48:50.626: 2023-10-25 01:48:50 (lasso.c/:182) Unable to load default xmlsec-crypto library. Make surethat you have it installed and check shared libraries path(LD_LIBRARY_PATH) environment variable.

func=xmlSecCryptoDLLibraryCreate:file=dl.c:line=146:obj=lt_dlopenext:subj=unknown:error=7:io function failed:filename=libxmlsec1-openssl

func=xmlSecCryptoDLGetLibraryFunctions:file=dl.c:line=498:obj=unknown:subj=xmlSecCryptoDLLibraryCreate:error=1:xmlsec library function failed:crypto=openssl

func=xmlSecCryptoDLLoadLibrary:file=dl.c:line=449:obj=unknown:subj=xmlSecCryptoDLGetLibraryFunctions:error=1:xmlsec library function failed:

**Comment by lsh123** [comment, marked as answer]
You are correct, it can't find libxmlsec1-openssl shared library:

> func=xmlSecCryptoDLLibraryCreate:file=dl.c:line=146:obj=lt_dlopenext:subj=unknown:error=7:io function failed:filename=libxmlsec1-openssl

For setting LD_LIBRARY_PATH for apache, check out this guidelines (it's more complicated than just setting it in the profile):

https://serverfault.com/questions/424997/how-to-set-ld-library-path-on-vps-red-hat-linux-to-be-visible-by-apache

Also make sure that dependency libraries (e.g. openssl itself) is also in the LD_LIBRARY_PATH. If you are still having problems, try to also set LTDL_LIBRARY_PATH as well in the same way (libltdl sometime is picky).
